src/checkers/inference/solver/util/StatisticRecorder.java

```diff
@@ -0,0 +1,88 @@
+package checkers.inference.solver.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Recorder for statistics.
+ * 
+ * @author jianchu
+ *
+ */
+public class StatisticRecorder {
+
+    public enum StatisticKey {
+        
+        /* Basic Info */
+        SLOTS_SIZE, 
+        CONSTRAINT_SIZE, 
+        GRAPH_SIZE,
+        CNF_VARIABLE_SIZE, 
+        CNF_CLAUSE_SIZE, 
+        LOGIQL_PREDICATE_SIZE, 
+        LOGIQL_DATA_SIZE, 
+        ANNOTATOIN_SIZE,
+        
+        /* Timing Info*/
+        GRAPH_GENERATION_TIME, 
+        OVERALL_PARALLEL_SOLVING_TIME,
+        OVERALL_SEQUENTIAL_SOLVING_TIME,
+        OVERALL_NOGRAPH_SOLVING_TIME,
+        SAT_SERIALIZATION_TIME,
+        SAT_SOLVING_TIME,
+        LOGIQL_SERIALIZATION_TIME, 
+        LOGIQL_SOLVING_TIME,
+    }
+
+    // Use atomic integer when back ends run in parallel.
+    public final static AtomicInteger satSerializationTime = new AtomicInteger(0);
+    public final static AtomicInteger satSolvingTime = new AtomicInteger(0);
+    private final static Map<StatisticKey, Long> statistic = new HashMap<StatisticKey, Long>();
+
+    static {
+        /* Initialize basic Info */
+        statistic.put(StatisticKey.SLOTS_SIZE, (long) 0);
```

Comment:
This initialization block seems initialize all statistic keys with an initial value 0 in the map. If that is the case, should we using a for-loop to initial the map? So we can avoid adding new statistic key while forget to add an initialize statement in this block.

E.g.

```java
for (StatisticKey key : StatisticKey.values()) {
    statistic.put(value, (long) 0);
}
```